cli/tests/stake.rs

```diff
@@ -972,3 +970,224 @@ fn test_stake_split() {
+        stake_account_pubkey,
+        lockup,
+        custodian: Some(read_keypair_file(&online_custodian_file).unwrap().into()),
+        sign_only: false,
+        signers: None,
+        blockhash_query: BlockhashQuery::default(),
+        nonce_account: None,
+        nonce_authority: None,
+        fee_payer: None,
+    };
+    process_command(&config).unwrap();
+    let stake_account = rpc_client.get_account(&stake_account_pubkey).unwrap();
+    let stake_state: StakeState = stake_account.state().unwrap();
+    let current_lockup = match stake_state {
+        StakeState::Initialized(meta) => meta.lockup,
+        _ => panic!("Unexpected stake state!"),
+    };
+    lockup.custodian = online_custodian_pubkey; // Default new_custodian is designated custodian
+    assert_eq!(current_lockup, lockup);
+
+    // Set custodian to offline pubkey
+    let lockup = Lockup {
+        unix_timestamp: 1581534573,
+        epoch: 203,
+        custodian: offline_pubkey,
+    };
+    config.command = CliCommand::StakeSetLockup {
+        stake_account_pubkey,
+        lockup,
+        custodian: Some(online_custodian.into()),
+        sign_only: false,
+        signers: None,
+        blockhash_query: BlockhashQuery::default(),
+        nonce_account: None,
+        nonce_authority: None,
+        fee_payer: None,
+    };
+    process_command(&config).unwrap();
+
+    // Create nonce account
+    let minimum_nonce_balance = rpc_client
+        .get_minimum_balance_for_rent_exemption(NonceState::size())
+        .unwrap();
+    let nonce_account = keypair_from_seed(&[1u8; 32]).unwrap();
+    let nonce_account_pubkey = nonce_account.pubkey();
+    let (nonce_keypair_file, mut tmp_file) = make_tmp_file();
+    write_keypair(&nonce_account, tmp_file.as_file_mut()).unwrap();
+    config.command = CliCommand::CreateNonceAccount {
+        nonce_account: read_keypair_file(&nonce_keypair_file).unwrap().into(),
+        seed: None,
+        nonce_authority: Some(offline_pubkey),
+        lamports: minimum_nonce_balance,
+    };
+    process_command(&config).unwrap();
+    check_balance(minimum_nonce_balance, &rpc_client, &nonce_account_pubkey);
+
+    // Fetch nonce hash
+    let account = rpc_client.get_account(&nonce_account_pubkey).unwrap();
+    let nonce_state: NonceState = account.state().unwrap();
+    let nonce_hash = match nonce_state {
+        NonceState::Initialized(_meta, hash) => hash,
+        _ => panic!("Nonce is not initialized"),
+    };
+
+    // Nonced offline set lockup
+    let lockup = Lockup {
+        unix_timestamp: 1581534576,
+        epoch: 222,
+        custodian: offline_pubkey,
+    };
+    config_offline.command = CliCommand::StakeSetLockup {
+        stake_account_pubkey,
+        lockup,
+        custodian: None,
+        sign_only: true,
+        signers: None,
+        blockhash_query: BlockhashQuery::None(nonce_hash, FeeCalculator::default()),
+        nonce_account: Some(nonce_account_pubkey.into()),
```

Comment:
This one is just an `Option<Pubkey>` no `.into()` required

Reply:
That's copy-pasta from `test_stake_split()` 😬

Reply:
:man_facepalming: Mind grabbing those too? :stuck_out_tongue:

Reply:
Done!